frontend/src/metabase-lib/parameters/utils/template-tags.ts

```diff
@@ -30,27 +30,36 @@ export function getTemplateTagParameterTarget(
     : ["variable", ["template-tag", tag.name]];
 }
 
-export function getTemplateTagParameter(tag: TemplateTag): ParameterWithTarget {
+export function getTemplateTagParameter(
+  tag: TemplateTag,
+  config?: ParameterConfig,
+): ParameterWithTarget {
   return {
     id: tag.id,
     type: tag["widget-type"] || getTemplateTagType(tag),
     target: getTemplateTagParameterTarget(tag),
     name: tag["display-name"],
     slug: tag.name,
     default: tag.default,
+    values_query_type: config?.values_query_type,
+    values_source_type: config?.values_source_type,
+    values_source_config: config?.values_source_config,
   };
 }
 
 // NOTE: this should mirror `template-tag-parameters` in src/metabase/api/embed.clj
 export function getTemplateTagParameters(
   tags: TemplateTag[],
+  parameters: Parameter[] = [],
 ): ParameterWithTarget[] {
+  const parametersById = _.indexBy(parameters, "id");
+
   return tags
     .filter(
       tag =>
         tag.type != null && (tag["widget-type"] || tag.type !== "dimension"),
     )
-    .map(getTemplateTagParameter);
+    .map(tag => getTemplateTagParameter(tag, parametersById[tag.id]));
```

Comment:
When re-generating parameters based on `template-tags`, we need to preserve new settings set in the `parameters` only.